src/kinetics/ReactionFactory.cpp

```diff
@@ -22,141 +22,204 @@ std::mutex ReactionFactory::reaction_mutex;
 ReactionFactory::ReactionFactory()
 {
     // register elementary reactions
-    reg("elementary", []() { return new ElementaryReaction(); });
+    reg("elementary", [](const AnyMap& node, const Kinetics& kin) {
+        return new ElementaryReaction3(node, kin);
+    });
     addAlias("elementary", "arrhenius");
     addAlias("elementary", "");
-    reg_XML("elementary",
-            [](Reaction* R, const XML_Node& node) {
-                setupElementaryReaction(*(ElementaryReaction*)R, node);
-            });
-    reg_AnyMap("elementary",
-               [](Reaction* R, const AnyMap& node, const Kinetics& kin) {
-                   setupElementaryReaction(*(ElementaryReaction*)R, node, kin);
-               });
+
+    // register elementary reactions (old framework)
+    reg("elementary-legacy", [](const AnyMap& node, const Kinetics& kin) {
+        Reaction* R = new ElementaryReaction2();
+        if (!node.empty()) {
+            setupElementaryReaction(*(ElementaryReaction2*)R, node, kin);
```

Comment:
You can avoid the cast by just doing:
```suggestion
        ElementaryReaction2* R = new ElementaryReaction2();
        if (!node.empty()) {
            setupElementaryReaction(*R, node, kin);
```